src/main/java/eu/bbmri/eric/csit/service/negotiator/service/NegotiationServiceImpl.java

```diff
@@ -86,20 +97,17 @@ public NegotiationDTO create(NegotiationCreateDTO negotiationBody, Long creatorI
           "One or more negotiationBody object is already assigned to another negotiation");
     }
 
-    // Gets the Role entity. Since this is a new negotiation, the person is the CREATOR of the negotiation
-    Role role = roleRepository.findByName("RESEARCHER")
-        .orElseThrow(EntityNotStorableException::new);
-
-    // Gets the person and associated roles
     Person creator =
-        personRepository.findDetailedById(creatorId).orElseThrow(EntityNotStorableException::new);
+        personRepository.findById(creatorId).orElseThrow(EntityNotStorableException::new);
+    addPersonToNegotiation(creator, negotiationEntity, "RESEARCHER");
 
-    // Ceates the association between the Person and the Negotiation
-    PersonNegotiationRole personRole = new PersonNegotiationRole(creator, negotiationEntity, role);
-
-    // Updates person and negotiation with the person role
-    creator.getRoles().add(personRole);
-    negotiationEntity.getPersons().add(personRole);
+    // TODO: Change how the resources are taken
+    Resource res = requests.get(0).getResources().stream().findFirst().get();
+    Set<Person> owners = res.getPersons();
+    //update the role for all owners
+    for (Person owner : owners) {
+      addPersonToNegotiation(owner, negotiationEntity, "REPRESENTATIVE");
+    }
```

Comment:
You can remove this because as I said this is already implemented in the hnadling of Perun groups: look at https://github.com/BBMRI-ERIC/negotiator-v3/blob/5ebdd1282203f60ec37d1167052647eb474eb280/src/main/java/eu/bbmri/eric/csit/service/negotiator/api/controller/v3/NegotiationController.java#L94C17-L94C17

Reply:
Yes, I'm working on it, I'll made another update to the PR

Reply:
okay

Reply:
Fixed in 9e79a341c22c9d82b6f49666b7b974893587a33c . Notice that, according to this new flow, that has direct impact in the information we get in the frontend's negotiation detail page, because there we don't have the information of the Representative in the summary table of the page